Fix 59138 - Enforce pywinrm v0.3.0 or higher

xeacott opened the pull request:
### What does this PR do?
Enforces pywinrm to be version 0.3.0 or higher.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/59138

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltstack.com/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

richardsimko committed:
Added SSH key specification when creating VM

richardsimko committed:
Added changelog docs

richardsimko committed:
Added tests

richardsimko committed:
Added support for firewall groups. Fixes #53677

richardsimko committed:
Added docs on SSH keys

richardsimko committed:
Linter fixes

richardsimko committed:
Fixed filename

me-coder committed:
#58747: Request addition of --addtag

#58747: Request addition of --addtag as valid kwarg to linux_lvm module

Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

Ch3LL committed:
Add vultr create tests for firewall and ssh keys

Commit:
Changelog added for #58747

Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

Commit:
Add missing parameters for vgcreate

Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

Commit:
Fixed modules/timezone.py to work in Slackware

Commit:
Added changelog entry

twangboy committed:
Pass port to smb connection

twangboy committed:
Add changelog

twangboy committed:
Add tests

twangboy committed:
Fix some black stuff

twangboy committed:
Fix more black

Commit:
init before lint

Commit:
init before lint

xeacott committed:
Merge branch 'master' into pywinrm_fix

Commit:
add changelog file

Commit:
add changelog file

xeacott committed:
Fix lint

xeacott committed:
Fix lint

xeacott committed:
Add SkipTest incase winrm is not found to not fail test.

Commit:
Add better message and var for version

Commit:
Add better message and var for version

Commit:
Merge branch 'pywinrm_fix' of github.com:xeacott/salt into pywinrm_fix

Commit:
change for consistency

Commit:
change for consistency

xeacott committed:
Fix pre-commit

xeacott committed:
Fix pre-commit

oeuftete commented on salt/utils/cloud.py:
```diff
@@ -1205,7 +1216,10 @@ def deploy_windows(
     if use_winrm and not HAS_WINRM:
-        log.error("WinRM requested but module winrm could not be imported")
+        log.error(
+            f"WinRM requested but module winrm could not be imported."
```
Is this safe at the moment, with f-strings needing python 3.6 or higher?

xeacott replied:
Probably not... Getting so used to using them I am forgetful to remove them. 😄

xeacott committed:
remove f string

xeacott committed:
remove f string

twangboy committed:
Convert vals to bytes

twangboy committed:
Fix tests

twangboy committed:
Add changelog

twangboy committed:
Allow use of query parameters in cmd.script source url

twangboy committed:
Add changelog

twangboy committed:
Fix pre-commit

xeacott committed:
Merge branch 'master' into pywinrm_fix

xeacott committed:
Merge branch 'master' into pywinrm_fix

xeacott committed:
re run pre commit to fix commas

xeacott committed:
re run pre commit to fix commas

Ch3LL commented on salt/utils/cloud.py:
```diff
@@ -1205,7 +1216,10 @@ def deploy_windows(
     if use_winrm and not HAS_WINRM:
-        log.error("WinRM requested but module winrm could not be imported")
+        log.error(
+            "WinRM requested but module winrm could not be importe. Ensure you are using version %s or higher.",
```
```suggestion
            "WinRM requested but module winrm could not be imported. Ensure you are using version %s or higher.",
```

xeacott replied:
Nice catch

Ch3LL commented:
Looks like your tests are not running. pywinrm is only included in the cloud integration requirement tests.  (https://jenkins.saltproject.io/job/pr-windows2019-py3-pytest/job/PR-59350/11/testReport/tests.unit.utils.test_cloud/CloudUtilsTestCase/test_winrm_pinnned_version/) Could you migrate these tests to tests/pytests/unit/ and make sure winrm is installed as a test requirement for unit tests

xeacott committed:
add pywinrm to ci for tests and fix mis spelling

xeacott committed:
add pywinrm to ci for tests and fix mis spelling

xeacott committed:
Merge branch 'master' into pywinrm_fix

Ch3LL commented:
There are some test failures that need to be cleaned up.

twangboy committed:
improve changelog

twangboy committed:
Fix pre-commit

twangboy committed:
Fix more pre-commit

twangboy committed:
Fix more pre-commit

Commit:
convert test to pytest

Commit:
convert test to pytest

Commit:
convert test to pytest

Commit:
convert test to pytest

xeacott committed:
pre-commit

xeacott committed:
pre-commit

Commit:
remove previous test file

xeacott committed:
Merge branch 'master' into pywinrm_fix

Commit:
Added timezone unit tests for Slackware

- Updated test_get_zone_os_family_allbsd_nilinuxrt to include Slackware
- Created test_set_zone_os_family_nilinuxrt for NILinuxRT
- Created test_set_zone_os_family_allbsd_slackware for FreeBSD, OpenBSD,
  NetBSD and Slackware
- Created tests from set_hwclock and get_hwclock for Slackware

s0undt3ch committed:
Update release notes

ScriptAutomate committed:
Add imagemagick for man pages gen

ScriptAutomate committed:
Trigger man pages gen

ScriptAutomate committed:
Remove test trigger of man page gen

dwoz committed:
Supress noisy warnings

Supress noisy warnings when running with very old pyzmq versions.
Fixes: #50327

s0undt3ch committed:
Drop Py2 and six on salt/cache/etcd_cache.py

s0undt3ch committed:
Drop Py2 and six on salt/cloud/clouds/gogrid.py

s0undt3ch committed:
Drop Py2 and six on salt/cache/mysql_cache.py

s0undt3ch committed:
Drop Py2 and six on salt/cloud/clouds/clc.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/bamboohr.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/apkpkg.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/aws_sqs.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/aix_group.py

s0undt3ch committed:
Drop Py2 and six on salt/cloud/clouds/scaleway.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/dockermod.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/baredoc.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/genesis.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/etcd_mod.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/gcp_addon.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/groupadd.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/jira_mod.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/nxos_upgrade.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/lxc.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/introspect.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/netscaler.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/iwtools.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/pagerduty.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/pacmanpkg.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/pagerduty_util.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/openvswitch.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/postfix.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/systemd_service.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/system.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/pw_group.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/splunk.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/s3.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/sysmod.py

s0undt3ch committed:
Drop Py2 and six on salt/runners/state.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/zabbix.py

s0undt3ch committed:
Drop Py2 and six on salt/states/github.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/zenoss.py

s0undt3ch committed:
Drop Py2 and six on salt/modules/telemetry.py

s0undt3ch committed:
Drop Py2 and six on salt/sdb/confidant.py

s0undt3ch committed:
Drop Py2 and six on salt/states/network.py

s0undt3ch committed:
Drop Py2 and six on salt/states/splunk.py

s0undt3ch committed:
Drop Py2 and six on salt/states/netusers.py

s0undt3ch committed:
Drop Py2 and six on salt/utils/etcd_util.py

Commit:
Merge branch 'pywinrm_fix' of github.com:xeacott/salt into pywinrm_fix

xeacott commented:
Yikes. Okay, just gunna close and reopen this.
